server/container_create_linux.go

```diff
@@ -569,35 +506,15 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 		if !isInCRIMounts("/sys", containerConfig.GetMounts()) {
 			specgen.RemoveMount("/sys")
 			specgen.RemoveMount("/sys/fs/cgroup")
```

Comment:
considering we call RemoveMount for effectively every call to add mount, I think it makes sense to add it to `AddMountToSpec` as well.

we'd still want this RemoveMount() call, because it wouldn't be removed when we remove `/sys`, but most others will be deduplicated

Reply:
this being the one that removes `/sys/fs/cgroup`, if that wasn't clear :slightly_smiling_face:

Reply:
it looks to me like there's quite a few places where we don't remove mounts :thinking: 
Could we call RemoveMount with an empty string to remove nothing, or maybe overload the function with another parameter ?

Reply:
I can't think of any case where calling RemoveMount before adding the mount would be problematic. The new mount should always replace the old, anyway

Reply:
Places like this though ?